src/rust/plugin-registry/src/server/deploy_plugin.rs

```diff
@@ -32,19 +32,39 @@ pub async fn deploy_plugin(
     db_client: &PluginRegistryDbClient,
     plugin: PluginRow,
     plugin_bucket_owner_id: &str,
+    plugin_bucket_name: &str,
+    plugin_bootstrap_container_image: &str,
+    plugin_execution_container_image: &str,
 ) -> Result<(), PluginRegistryServiceError> {
     // --- Convert HCL to JSON Job model
     let job_name = "grapl-plugin"; // Matches what's in `plugin.nomad`
     let job = {
         let job_file_hcl = static_files::PLUGIN_JOB;
+        let kernel_artifact_url = format!(
+            "https://{bucket}.s3.amazonaws.com/{key}",
+            bucket = plugin_bucket_name,
+            key = "kernel/v0.tar.gz",
+        );
+        let rootfs_artifact_url = format!(
+            "https://{bucket}.s3.amazonaws.com/{key}",
+            bucket = plugin_bucket_name,
+            key = "rootfs/v0.rootfs.tar.gz",
+        );
```

Comment:
also, in general I find that lifting the string interpolation out of nomad has been beneficial, so I made this change too

Reply:
👍 for removing as much string interpolation from the HCL file as possible, in favor of injecting "complete" values.

Is there any incidental complexity that we're introducing by having values injected from our Pulumi code and now from our Rust code? Or are the data we're manipulating here in the Rust code at a level that isn't available to Pulumi (e.g., dependent on a specific tenant or plugin execution)?

Reply:
> incidental complexity that we're introducing by having values injected from our Pulumi code and now from our Rust code
I'd argue that more entrypoints for deploying-to-nomad is, inherently, complexity (we've already seen devs having trouble grasping the deploying-to-nomad-from-shell-for-local-services versus deploying-to-nomad-from-pulumi)

> are the data we're manipulating here in the Rust code at a level that isn't available to Pulumi (e.g., dependent on a specific tenant or plugin execution)?
For the rootfs/kernel - I don't actually know! - but there are other variables (primarily, `plugin: PluginRow`) that do change between specific plugins.

If the question is "could we inject these two variables at the Pulumi level as well", I _think_ so